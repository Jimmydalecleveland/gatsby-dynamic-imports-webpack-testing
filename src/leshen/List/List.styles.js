import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const List = styled.ul(
  ({ as }) => css`
    display: inline-block;
    text-align: left; /* for centered lists */
    list-style: none;

    ${as === 'ol' &&
    `
      counter-reset: number-list-style;

      > li {
        counter-increment: number-list-style;
        &:before {
          content: counter(number-list-style)'.';
        }

        svg {
          display: none;
        }
      }
    `}
  `
)
export const ListItem = styled.li`
  margin-bottom: ${({ theme }) => theme.spacing.xxs};
  display: flex;
  align-items: flex-start;

  > div {
    margin-left: ${({ theme }) => theme.spacing.xxs};
    flex: 1;
  }

  > svg {
    margin: 0;
  }

  h5 {
    margin-bottom: 0;
  }

  p {
    margin: 0;
  }
`
