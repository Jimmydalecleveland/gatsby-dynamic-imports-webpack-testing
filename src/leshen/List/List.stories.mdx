import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'
import { Done, Fingerprint } from 'emotion-icons/material'

import List, { ticketID } from '.'
import ListItem from './ListItem'
import ComponentProgress from '../ComponentProgress'

<Meta title='Components|List' component={List} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# List
By default the list renders as a `ul`.
<Preview>
  <Story name="unordered">
    <List>
      <ListItem content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur." />
      <ListItem content="Two" />
    </List>
  </Story>
</Preview>

## Props
#### List
<Props of={List} />

#### ListItem
<Props of={ListItem} />

## Ordered List
<Preview>
  <Story name="ordered">
    <List type="ordered">
      <ListItem content="One" />
      <ListItem content="Two" />
    </List>
  </Story>
</Preview>

## Icons
<Preview>
  <Story name="icons">
    <List>
      <ListItem icon={<Done color="mediumaquamarine" />} content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur." />
      <ListItem icon={<Done color="mediumaquamarine" />} content="Two" />
    </List>
  </Story>
</Preview>

#### Icon and multiple child content
<Preview>
  <Story name="icon with multiple child content">
    <List>
      <ListItem icon={<Fingerprint size="50px" />}>
        <div>
          <strong>Press and hold here</strong>
        </div>
        <em>To pay me $1,500 smakaroos</em>
      </ListItem>
    </List>
  </Story>
</Preview>
