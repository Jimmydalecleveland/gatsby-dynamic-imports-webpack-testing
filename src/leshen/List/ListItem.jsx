import React from 'react'
import PropTypes from 'prop-types'
import { Lens as Disc } from 'emotion-icons/material'

import * as Styled from './List.styles'

const ListItem = ({ icon, content, children, ...props }) => (
  <Styled.ListItem icon={icon && true} {...props}>
    {icon && icon}
    {content && <div>{content}</div>}
    {children && <div>{children}</div>}
  </Styled.ListItem>
)

ListItem.defaultProps = {
  /**  Disc is the "bullet" icon we default to for consistently using an SVG icon */
  icon: <Disc size="0.5em" style={{ marginTop: '0.45em' }} />,
  content: null,
  children: null,
}

ListItem.propTypes = {
  /** Defaults to a bullet icon (Disc) */
  icon: PropTypes.element,
  // TODO: Should this exist? Why not just pass everything as children?
  content: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]),
}

export default ListItem
