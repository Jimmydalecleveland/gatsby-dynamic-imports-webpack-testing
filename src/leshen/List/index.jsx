import React from 'react'
import PropTypes from 'prop-types'
import * as Styled from './List.styles'

const List = ({ type, children, ...props }) => {
  let listTag = 'ul'
  if (type !== 'unordered') {
    listTag = 'ol'
  }

  return (
    <Styled.List as={listTag} {...props}>
      {children}
    </Styled.List>
  )
}

List.defaultProps = {
  type: 'unordered',
}

List.propTypes = {
  type: PropTypes.oneOf(['ordered', 'unordered']),
  children: PropTypes.node.isRequired,
}

export default List

export const ticketID = 'CMS-168'
