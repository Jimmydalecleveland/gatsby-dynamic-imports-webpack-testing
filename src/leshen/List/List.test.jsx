import React from 'react'
import { render } from 'setup-tests'
import List from '.'
import ListItem from './ListItem'

const Done = () => (
  <svg
    data-testid="svgicon"
    viewBox="0 0 24 24"
    aria-hidden="true"
    focusable="false"
    role="img"
    fill="currentColor"
  >
    <path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
  </svg>
)

describe('List', () => {
  test('List should render', () => {
    const { container } = render(
      <List>
        <ListItem content="One" />
        <ListItem>
          <p>Two</p>
        </ListItem>
      </List>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('List renders with an optional icon', () => {
    const { container, getAllByTestId } = render(
      <List>
        <ListItem icon={<Done />} content="One" />
        <ListItem icon={<Done />}>
          <p>Two</p>
        </ListItem>
      </List>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(getAllByTestId('svgicon'))
  })
})
