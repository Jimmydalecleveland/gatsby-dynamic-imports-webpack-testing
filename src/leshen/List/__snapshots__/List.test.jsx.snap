// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List renders with an optional icon 1`] = `
.emotion-4 {
  display: inline-block;
  text-align: left;
  list-style: none;
}

.emotion-0 {
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 > div {
  margin-left: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 > svg {
  margin: 0;
}

.emotion-0 h5 {
  margin-bottom: 0;
}

.emotion-0 p {
  margin: 0;
}

<ul
  class="emotion-4 emotion-5"
>
  <li
    class="emotion-0 emotion-1"
  >
    <svg
      aria-hidden="true"
      data-testid="svgicon"
      fill="currentColor"
      focusable="false"
      role="img"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
    <div>
      One
    </div>
  </li>
  <li
    class="emotion-0 emotion-1"
  >
    <svg
      aria-hidden="true"
      data-testid="svgicon"
      fill="currentColor"
      focusable="false"
      role="img"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
    <div>
      <p>
        Two
      </p>
    </div>
  </li>
</ul>
`;

exports[`List List should render 1`] = `
.emotion-4 {
  display: inline-block;
  text-align: left;
  list-style: none;
}

.emotion-0 {
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 > div {
  margin-left: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 > svg {
  margin: 0;
}

.emotion-0 h5 {
  margin-bottom: 0;
}

.emotion-0 p {
  margin: 0;
}

<ul
  class="emotion-4 emotion-5"
>
  <li
    class="emotion-0 emotion-1"
  >
    <span>
      Lens icon Mock
    </span>
    <div>
      One
    </div>
  </li>
  <li
    class="emotion-0 emotion-1"
  >
    <span>
      Lens icon Mock
    </span>
    <div>
      <p>
        Two
      </p>
    </div>
  </li>
</ul>
`;
